### PR TITLE
Path: Refactor/pylint warning cleanup

### PR DESCRIPTION
**Pull request by mlampert**
Addresses or silences all pylint3 warnings for the core modules. Caught some bugs and makes some questionable coding habits visible.

Some modules still require cleanup - but this is already a massive PR (and maintaining it outside master is error prone) so I would prefer getting it merged in stages.

### CODE REVIEW

File: src/Mod/Path/PathScripts/PathDressupDogbone.py [hunk 1]
```diff
@@ -35,8 +35,6 @@
 
 from PySide import QtCore
 
-"""Dogbone Dressup object and FreeCAD command"""
```
Comment (Russ4262) [inline src/Mod/Path/PathScripts/PathDressupDogbone.py]: __doc__ = "Dogbone Dressup object and FreeCAD command"

File: src/Mod/Path/PathScripts/PathDressupDragknife.py [hunk 2]
```diff
@@ -30,7 +30,7 @@
 import DraftVecUtils as D
 import PathScripts.PathUtils as PathUtils
 
-"""Dragknife Dressup object and FreeCAD command"""
+__doc__ = """Dragknife Dressup object and FreeCAD command"""
```
Comment (Russ4262) [inline src/Mod/Path/PathScripts/PathDressupDragknife.py]: __doc__ = "Dragknife Dressup object and FreeCAD command"

File: src/Mod/Path/PathScripts/PathDressupHoldingTags.py [hunk 3]
```diff
@@ -36,8 +36,6 @@
 from PathScripts.PathUtils import waiting_effects
 from PySide import QtCore
 
-"""Holding Tags Dressup object and FreeCAD command"""
```
Comment (Russ4262) [inline src/Mod/Path/PathScripts/PathDressupHoldingTags.py]: __doc__ = "Holding Tags Dressup object and FreeCAD command"

File: src/Mod/Path/PathScripts/PathDressupHoldingTags.py [hunk 4]
```diff
@@ -177,13 +183,13 @@ def createSolidsAt(self, z, R):
...
-            return list(filter(lambda pt: pt.z >= self.bottom() and pt.z <= self.top(), pts))
+            return list([pt for pt in pts if pt.z >= self.bottom() and pt.z <= self.top()])
         if type(face.Surface) == Part.Plane:
             PathLog.track("it's a plane, checking R")
             c = face.Edges[0].Curve
             if (type(c) == Part.Circle):
-                return list(filter(lambda pt: (pt - c.Center).Length <= c.Radius or PathGeom.isRoughly((pt - c.Center).Length, c.Radius), pts))
-        print("==== we got a %s" % face.Surface)
+                return list([pt for pt in pts if (pt - c.Center).Length <= c.Radius or PathGeom.isRoughly((pt - c.Center).Length, c.Radius)])
+        PathLog.error("==== we got a %s" % face.Surface)
```
Comment (Russ4262) [inline src/Mod/Path/PathScripts/PathDressupHoldingTags.py]: Just a question: Which syntax is preferred moving forward for Python 3.7 and beyond?
1.)        PathLog.error("==== we got a %s" % face.Surface)
2.)        PathLog.error("==== we got a {}".format(face.Surface))

File: src/Mod/Path/PathScripts/PathDressupLeadInOut.py [hunk 5]
```diff
@@ -34,7 +34,7 @@
 
 from PySide import QtCore
 
-"""LeadInOut Dressup MASHIN-CRC USE ROLL-ON ROLL-OFF to profile"""
+__doc__ = """LeadInOut Dressup MASHIN-CRC USE ROLL-ON ROLL-OFF to profile"""
```
Comment (Russ4262) [inline src/Mod/Path/PathScripts/PathDressupLeadInOut.py]: __doc__ = "LeadInOut Dressup MASHIN-CRC USE ROLL-ON ROLL-OFF to profile"

**Comment by mlampert** [comment]
format() is probably the correct way of dealing with strings. With bulk
cleanups you mostly end up just dealing with the problem at hand so I
just converted all print() statements to PathLog.

**Pull request merged**